dev/planning-example.py

```diff
@@ -0,0 +1,265 @@
+
+# It's also possible to initialize using astropy objects and other
+# direct entities
+from astropy.coordinates.angles import Longitude, Latitude
+import astropy.units as u
+import pytz
+
+obs = entity.Observer(name='Keck 1',
+                      longitude=Longitude(155.4750 * u.degree),
+                      latitude=Latitude(19.8264 * u.degree)
+                      elevation=4160 * u.m,
+                      pressure=615 * u.hPa,
+                      relative_humidity=0.11,
+                      temperature=0.0 * u.C,
+                      timezone=pytz.timezone('US/Hawaii'),
+                      description="W.M. Keck 1 Telescope on Mauna Kea, Hawaii")
+
+# It would also be desirable to be able to have a small database of
+# common telescopes.  Maybe this can at first simply take the form of
+# a python module:
+from astroplan import sites
+
+obs = sites.get_observer('keck1')
+
+# Environmental conditions should be updatable.  Suggest using keyword
+# parameters (this would make it easy to update an internal AltAz coordinate):
+#
+obs.set_env_conditions(pressure=600 * u.hPa, relative_humidity=0.2,
+                       temperature=10 * u.C)
+
+# It's easy to construct date/times from the observer, since it knows about
+# its timezone.  These dates can be passed directly to target check methods.
+
+# dates default to midnight if not otherwise specified
+day = obs.get_date("2015-05-01")
+
+# Can get more and more specific
+t1 = obs.get_date("2015-05-01 21:00")
+t1 = obs.get_date("2015-05-01 21:33:41")
+
+# default is current time
+cur_time = obs.get_date()
+
+# Dates are assumed to be in the observer's time zone unless a time zone
+# specifier is given
+t2_utc = obs.get_date("2015-05-01 21:33:41 UTC")
+
+"""
+We can ask about typical times of interest at this observing position
+Returned dates are assumed to be in the observer's time zone for convenience.
+The date parameter, if given, can be an astropy Time or a datetime instance.
+If no parameter is given, the current date/time is assumed.
+"""
+
+# sunset
+obs.sunset(date=day)
+
+# sunrise
+obs.sunrise(date=day)
+
+# moon rise
+obs.moon_rise(date=day)
+
+# moon set
+obs.moon_set(date=day)
+
+# evening (nautical) twilight
+obs.evening_twilight_12(date=day)
+
+# evening (civil) twilight
+obs.evening_twilight_18(date=day)
+
+# morning (nautical) twilight
+obs.morning_twilight_12(date=day)
+
+# morning (civil) twilight
+obs.morning_twilight_18(date=day)
+
+# what is the moon illumination?
+# returns a float, which is percentage of the moon illuminated
+obs.moon_illumination(date=t1)
+
+# what is the moon altitude and azimuth?
+# returns an altaz coordinate
+obs.moon_position(date=t1)
+
+# ==============
+# Target objects
+# ==============
+'''
+A target object defines a single observation target, with coordinates
+and an optional name.
+'''
+from astroplan.entity import SiderealTarget
+
+# Define a target.
+t1 = SiderealTarget(name='Polaris', ra='02:31:49.09', dec='+89:15:50.8')
```

Comment:
One additional thought.  If it is desirable to use `SkyCoord` directly (e.g. for vectorizing), then maybe we can simply provide convenience methods for creating a SkyCoord _with_ the name attached (or whatever additional metadata).  Perhaps if this is useful enough (metadata in `SkyCoord`s) such capability should make its way into `SkyCoord` eventually.